nixos/modules/services/monitoring/prometheus/exporters/sql.nix

```diff
@@ -0,0 +1,28 @@
+{ config, lib, pkgs, options }:
+with lib;
+let
+  cfg = config.services.prometheus.exporters.sql;
+in
+{
+  extraOpts = {
+    configFile = lib.mkOption {
```

Comment:
I think it would be more viable to provide a `settings`-option (as described in [RFC#42](https://github.com/nixos/rfcs/pull/42)). In the end you can generate a YAML from this using `builtins.toJSON`.

Also, with this approach it would be possible to override a single config value without replacing the entire `configFile`.

Reply:
"Overriding a single option" does not really work for sql_exporter as it *requires* a config, unlike some other exporters. However, it should not be too hard to add options according to [the config schema](https://github.com/justwatchcom/sql_exporter#configuration) and have an assertion that either `configFile` or `jobs` be present.

Reply:
What do you mean with "requires a config". You can generate the file's content by serializing a Nix attr-set.

Reply:
Sorry if I was unclear. I meant that 

> override a single config value without replacing the entire configFile

is something that makes sense with something like a `bind` or `dovecot2` exporter where it works out of the box because you can just have a default config that points to `localhost` or something. But with sql_exporter there is no reasonable default where you could just say `enable = true;` and have it working - it does not make sense to define (*override*) just one option like DB connection string, as you also need to define queries, data points columns, labels, etc. for it to meaningfully function at all.

But anyway, that was just a side note. I should get around to making proper, typed options for this module in the coming days.